solr/core/src/java/org/apache/solr/servlet/SolrDispatchFilter.java

```diff
@@ -227,6 +234,44 @@ private void setupJvmMetrics(CoreContainer coresInit)  {
     }
   }
 
+  class NoOpThreadsDeadlockDetector extends ThreadDeadlockDetector {
```

Comment:
can this be put into a separate class. `SolrDispatchFilter` is already unnecessarily complex